### ISSUE TITLE
Compile to WebAssembly?

### ISSUE THREAD

**Issue by mansellan** [WaynePhillipsEA/twinbasic#570]
**Is your feature request related to a problem?**
Not a problem for right now. But as twinBASIC expands into cross-platform, should it consider WebAssembly?

**Describe the solution you'd like**
A new compilation target for WebAssembly.

**Describe alternatives you've considered**
Not compiling to WebAssembly 😜 

**Additional context**
WebAssembly is "the new shiny" - it may or may not have legs. At the moment, it seems to be gaining traction and getting lots of fans. I have no idea how difficult support would be, but it seems to me that tewnBASIC's tiny "runtime" could be a big advantage? .Net / Mono have struggled getting their VM-based runtime acceptably small.

**Comment by WaynePhillipsEA** [comment]
Yes, I totally agree, this should be considered.  This has been in the back of my mind for some time.  I (very briefly) mentioned it in https://github.com/WaynePhillipsEA/twinbasic/issues/271#issuecomment-924612541.

**Comment by mburns08109** [comment]
Could someone add a link to relevant reading up on this subject (preferably articles that won't leave inveterate VBx-ers in the dust)?

**Comment by mansellan** [comment]
[This seems pretty good](https://blogs.remobjects.com/2019/07/18/webassembly-a-primer/).

Not had dealings with it myself (so please don't quote me!), but WebAssembly purports to be a high-speed execution engine for the browser (technically a VM). IIUC, it's low-level, synchronous, stack-based, and easily translated to native instructions by the browser. Everything that JavaScript is not.

You don't program in WebAssembly directly, rather you target it from a higher level language. Which means you can be a front-end web developer without having to know anything at all about JavaScript (win!). Many languages have started supporting it, due to it's promise: small download size, fast execution, code-in-any-language.

**Comment by mburns08109** [comment]
> [This seems pretty good](https://blogs.remobjects.com/2019/07/18/webassembly-a-primer/).
> 
> Not had dealings with it myself (so please don't quote me!), but WebAssembly purports to be a high-speed execution engine for the browser (technically a VM). IIUC, it's low-level, synchronous, stack-based, and easily translated to native instructions by the browser. Everything that JavaScript is not.
> 
> You don't program in WebAssembly directly, rather you target it from a higher level language. Which means you can be a front-end web developer without having to know anything at all about JavaScript (win!). Many languages have started supporting it, due to it's promise: small download size, fast execution, code-in-any-language, and plugin-free.

(looks up as that content flew past me a few thousand feet overhead).
mmmNope. I kinda recognized some parts of that, but....just nope.

**Comment by EduardoVB** [comment]
I guess perhaps it would be the most important feature to have in the future.
If WebAssembly gets established as a browser technology, as it seems it will be (or is already?), in my (limited now) understanding I think that JavaScript reign is coming to an end.

Browser based apps will continue gaining ground gradually, and they are by default cross platform.
Having the ability to program such apps in VB+ (aka tB) would be more than desirable.
If I would be able today to program web apps in tB, I think I would be already using that technology.

Now, everybody is forced to use JavaScript (or a framework based on it).
I think it could be a big booster for tB. Of course I mean in the future.
